content/departments/product-engineering/engineering/cloud/devops/resize-disks-in-statefulset.md

```diff
@@ -0,0 +1,73 @@
+# Resize disks in GKE StatefulSet
+
+Important: this change will caused recreation on all StatefulSet's pods, which can cause donwtime if application is not high-available.
+
+### Steps to resize StatefulSet persistent disks
+
+1. Check the actual disks sizes in GCP
+
+```
+gcloud compute disks list
+```
+
+2. List existing persistent volumes and corresponding persistent volume claims:
+
+```
+kubectl get pv
+kubectl get pvc -n prod
+```
+
+3. Ensure that the storage class of the volumes for resizing has set `AllowVolumeExpansion=True`
+
+```
+kubectl describe sc sourcegraph | grep AllowVolumeExpansion
+# should be: AllowVolumeExpansion:  True
+```
+
+4. Ensure that storage class of the volumes for resizing has set `ReclaimPolicy=Retain`
+
+```
+kubectl describe sc sourcegraph | grep ReclaimPolicy
+# should be: ReclaimPolicy: Retain
+```
+
+5. Modify persistent volume claims to have new size
+
+Note: do it for ALL persistent volumes used by StatefulSet!
+
+```
+kubectl patch pvc repos-gitserver-0 -n prod -p '{ "spec": { "resources": { "requests": { "storage": "250Gi" }}}}'
+```
+
+6. Ensure that all resized persistent volumes, persistent volume claims and google disks has new value (250Gi in this example):
+
+```
+kubectl get pv
+kubectl get pvc -n prod
+gcloud compute disks list
+```
+
+7. Delete StatefulSet without deleting the pods (`--cascade='orphan'`)
+
+Important: without `--cascade='orphan'` you will loose the data!
+
+```
+kubectl delete sts gitserver --cascade=orphan -n prod
+```
+
+8. Modify and deploy StatefulSet with extended disk sizes (change `storage` to `250Gi` in this example):
+
+- in [StatefulSet](https://k8s.sgdev.org/github.com/sourcegraph/deploy-sourcegraph-cloud/-/blob/base/gitserver/gitserver.StatefulSet.yaml?L148)
+- IN persistent volumes files: https://k8s.sgdev.org/github.com/sourcegraph/deploy-sourcegraph-cloud/-/tree/base/gitserver (all files with `repos-gitserver-[NUMBER].PersistentVolume.yaml` names)
+- deploy via [buildkite job](https://buildkite.com/sourcegraph/deploy-sourcegraph-cloud)
```

Comment:
Should we replace these links with `preview.sgdev.dev`?

Reply:
As soon as we have everything indexed in preprod yes, but now it still does not work: seems like `deploy-sourcegraph-cloud` is not indexed yet.